src/js/components/search/filters/budgetCategory/BudgetCategoryOCSearch.jsx

```diff
@@ -4,34 +4,61 @@
  */
 
 import React from 'react';
-import _ from 'lodash';
-import { objectClassDefinitions } from 'dataMapping/search/budgetCategory';
+import { objectClassDefinitions, objectClassDefinitionsGroups }
+    from 'dataMapping/search/budgetCategory';
 
 import PrimaryCheckboxType from 'components/sharedComponents/checkbox/PrimaryCheckboxType';
 
+const defaultProps = {
+    objectClassMapping: [
+        {
+            id: 'personnel_compensation_and_benefits',
+            name: 'Personnel compensation and benefits',
+            filters: objectClassDefinitionsGroups.personnel_compensation_and_benefits
+        },
+        {
+            id: 'contractual_services_and_supplies',
+            name: 'Contractual services and supplies',
+            filters: objectClassDefinitionsGroups.contractual_services_and_supplies
+        },
+        {
+            id: 'acquisition_of_assets',
+            name: 'Acquisition of assets',
+            filters: objectClassDefinitionsGroups.acquisition_of_assets
+        },
+        {
+            id: 'grants_and_fixed_changes',
+            name: 'Grants and fixed charges',
+            filters: objectClassDefinitionsGroups.grants_and_fixed_changes
+        },
+        {
+            id: 'other',
+            name: 'Other',
+            filters: objectClassDefinitionsGroups.other
+        }
+    ]
+};
+
 const propTypes = {
-    selectObjectClass: React.PropTypes.func,
-    objectClasses: React.PropTypes.object
+    objectClassMapping: React.PropTypes.arrayOf(React.PropTypes.object),
+    objectClasses: React.PropTypes.object,
+    selectObjectClass: React.PropTypes.func
 };
 
 export default class BudgetCategoryOCSearch extends React.Component {
     render() {
-        const objectClassItems = [];
-
-        Object.keys(objectClassDefinitions).forEach((key) => {
-            objectClassItems.push(
+        const objectClassItems = (
+            this.props.objectClassMapping.map((type, index) =>
                 <PrimaryCheckboxType
+                    {...type}
                     {...this.props}
-                    key={key}
-                    id={`award-${key}`}
+                    key={index}
```

Comment:
The `objectClassMapping` already has unique values that can be better utilized as `key`. Again, using an iterator index as `key` should be a last option.